### PR TITLE
Add Plugin status provider

### PR DESCRIPTION
**Pull request by lightglitch**

### CODE REVIEW

**Comment by decebals** [comment]
Can you please describe in few words your use case or the user case resolved by this PR?

**Comment by lightglitch** [comment]
I want the plugins enable/disable configuration to be fetch from a database, this allows the user to change this configuration in the interface.

File: pf4j/src/main/java/ro/fortsoft/pf4j/DefaultPluginManager.java [hunk 1]
```diff
@@ -658,6 +658,37 @@ protected PluginClasspath createPluginClasspath() {
     	return new PluginClasspath();
     }
 
+    protected PluginStatusProvider createPluginStatusProvider() {
+        return new PluginStatusProvider() {
```
Comment (decebals) [inline pf4j/src/main/java/ro/fortsoft/pf4j/DefaultPluginManager.java]: My idea is to move the default implementation in a `DefaultPluginStatusProvider`.

Comment (lightglitch) [inline pf4j/src/main/java/ro/fortsoft/pf4j/DefaultPluginManager.java]: Ok, I will update that.

File: pf4j/src/main/java/ro/fortsoft/pf4j/DefaultPluginManager.java [hunk 2]
```diff
@@ -658,6 +658,37 @@ protected PluginClasspath createPluginClasspath() {
...
+                } catch (IOException e) {
+                    log.error(e.getMessage(), e);
+                }
+                return disabledPlugins;
+            }
+        };
+    }
+
     protected boolean isPluginDisabled(String pluginId) {
     	if (enabledPlugins.isEmpty()) {
```
Comment (decebals) [inline pf4j/src/main/java/ro/fortsoft/pf4j/DefaultPluginManager.java]: I think that we can remove `enabledPlugins`, `disabledPlugins` from DefaultPluginManager and in `initialize` in pace of
```java
enabledPlugins = statusLists.getEnabledPlugins();
disabledPlugins = statusLists.getDisabledPlugins();
```

we can put 
```java
pluginStatusProvider = createPluginStatusProvider();
```

File: pf4j/src/main/java/ro/fortsoft/pf4j/PluginStatusProvider.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+/**
+ * @author Decebal Suiu
+ * @author Mário Franco
+ */
+public interface PluginStatusProvider {
+
+    /**
+     * Retrieves a list with plugin identifiers that should be only accepted by this manager
+     */
+    public List<String> getEnabledPlugins();
```
Comment (decebals) [inline pf4j/src/main/java/ro/fortsoft/pf4j/PluginStatusProvider.java]: Is it necessary to expose these methods? I see that is enough a `isPluginDisabled` method for `DefaultPluginManager.isPluginDisabled`.

Comment (lightglitch) [inline pf4j/src/main/java/ro/fortsoft/pf4j/PluginStatusProvider.java]: I only saw that now that you mentioned and I think it's the best option. 
I will clean up the code.

**Comment by lightglitch** [comment]
Updated.

**Comment by decebals** [comment]
:+1: Good job. Thanks

**Pull request merged**